### PR TITLE
staticanalysis/bot: Convert zero coverage analyzer to remote task

### PR DESCRIPTION
**Pull request by La0**

### CODE REVIEW

File: src/staticanalysis/bot/static_analysis_bot/coverage.py [hunk 1]
```diff
@@ -104,32 +100,26 @@ def as_phabricator_lint(self):
...
+    '''
+    route = 'project.releng.services.project.production.code_coverage_bot.latest'
+    artifacts = [
+        'public/zero_coverage_report.json',
+    ]
+
+    def parse_issues(self, artifacts, revision):
+        zero_coverage_files = {
+            file_info['name']
+            for artifact in artifacts.values()
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/coverage.py]: Why artifacts.values(), if artifacts is a list?

File: src/staticanalysis/bot/static_analysis_bot/workflows/remote.py [hunk 2]
```diff
@@ -78,11 +80,19 @@ def _in_group(dep_id):
...
+        # Add zero-coverage task
+        dependencies.append(ZeroCoverageTask)
+
         # Find issues and patches in dependencies
         issues = []
-        for dep_id in dependencies:
+        for dep in dependencies:
             try:
-                task = self.build_task(dep_id, tasks[dep_id])
+                if isinstance(dep, type):
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/workflows/remote.py]: Nit: isinstance(dep, AnalysisTask)?

File: src/staticanalysis/bot/static_analysis_bot/coverage.py (hunk 1 above)
Comment (marco-c) [reply src/staticanalysis/bot/static_analysis_bot/coverage.py]: Forget it, it's a dict.

**Comment by marco-c** [comment]
The artifact should always be available, but maybe to be 100% safe we could make the bot not crash when it is not present and just warn.

**Comment by La0** [comment]
I made the bot **not** crash when the route is missing (as shown in unit tests)

**Pull request merged**